[ACR] az acr connected-acr create/update/delete/list/show: new feature in beta.

rosanch opened the pull request:
Description
Adding Connected Registry code to the CLI

History Notes
[Component Name 1] az acr connected-acr: New beta feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

rosanch committed:
Connected ACR Template

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_format.py:
```diff
@@ -243,6 +251,29 @@ def _agentpool_format_group(item):
+        ('STATUS', _get_value(item, 'status')),
+        ('PARENT', _get_value(item, 'parent')),
+        ('LOGIN NAME', _get_value(item, 'loginName')),
+        ('AUTO UPDATE', _get_value(item, 'autoUpdate')),
```
Auto update is not supported for MVP, please remove

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_format.py:
```diff
@@ -243,6 +251,29 @@ def _agentpool_format_group(item):
+        ('LOGIN NAME', _get_value(item, 'loginName')),
+        ('AUTO UPDATE', _get_value(item, 'autoUpdate')),
+        ('LAST UPDATED', _get_value(item, 'lastUpdated')),
+        ('NEXT UPDATE', _get_value(item, 'nextUpdate')),
```
What property does this correspond to? `NextSyncTime` is an internal only property

rosanch replied:
Not sure yet. I'm following the cli-experience.md
https://msazure.visualstudio.com/AzureContainerRegistry/_git/functional-specs?path=%2FRegistry%2FConnected-ACR%2Fcli-experience.md&_a=preview

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_format.py:
```diff
@@ -243,6 +251,29 @@ def _agentpool_format_group(item):
+        ('PARENT', _get_value(item, 'parent')),
+        ('LOGIN NAME', _get_value(item, 'loginName')),
+        ('AUTO UPDATE', _get_value(item, 'autoUpdate')),
+        ('LAST UPDATED', _get_value(item, 'lastUpdated')),
```
Does this correspond to the `LastVersionUpdateTime`, `LastSyncTime`, or `LastActivityTime`?

rosanch replied:
That's my guess too. The list/show commands have quite a bit of work left to go, I added for now to match the Design spec

rosanch replied:
the cli experienced didn't specified so I set it to lastActivityTime for now

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_format.py:
```diff
@@ -243,6 +251,29 @@ def _agentpool_format_group(item):
+        ('NAME', _get_value(item, 'name')),
+        ('STATUS', _get_value(item, 'status')),
+        ('PARENT', _get_value(item, 'parent')),
+        ('LOGIN NAME', _get_value(item, 'loginName')),
```
@rosanch please validate that the dictionary entries here map to the correct property in the model. For instance, LOGIN NAME corresponds to `_get_value(item, 'loginServer', 'host')`

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_help.py:
```diff
@@ -1204,6 +1204,98 @@
+    text: |
+        az acr connecter-acr create --registry mycloudregistry.azurecr.io
+            --name Contoso-Mirror01-in-Wichita --repositories contoso-app/hello-world
+            --next_update "0 0 12 1/1 * ? *"
```
@rosanch the cron property corresponds to `--sync-schedule` flag

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_help.py:
```diff
@@ -1204,6 +1204,98 @@
+    text: |
+        az acr connecter-acr create --registry mycloudregistry.azurecr.io
+            --name Contoso-Mirror02-in-Wichita --repositories contoso-app/mycomponent
+            --sync-schedule "0 0 12 1/1 * ? *" --sync-window 4H --disable_auto_update
```
@rosanch we support 5-field crons only. For instance `"30 */1 * * *"` which translates to "At minute 30 past every hour"

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_help.py:
```diff
@@ -1204,6 +1204,98 @@
+        az acr connecter-acr create --registry mycloudregistry.azurecr.io
+            --mode mirror --parent Contoso-Edge01-in-Wichita
+            --name Contoso-Mirror02-in-Wichita --repositories contoso-app/mycomponent
+            --sync-schedule "0 0 12 1/1 * ? *" --sync-window 4H
```
@rosanch we support 5-field crons only. The equivalent of a schedule that syncs every day at midnight would be the expression `0 12 * * *`

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_help.py:
```diff
@@ -1204,6 +1204,98 @@
+  - name: Create a connected registry with DNS name 'edge01.contoso.internal' that has 'mycloudregistry.azurecr.io' as a parent.
+    text: |
+        az acr connecter-acr create --registry mycloudregistry.azurecr.io 
+            --name Contoso-Edge01-in-Wichita --repositories "contoso-app/* contoso-service/*"
```
@rosanch please remove wildcard from repositories. For MVP we ask users to explicitly list each repository

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_help.py:
```diff
@@ -1204,6 +1204,98 @@
+    text: |
+        az acr connecter-acr update --registry mycloudregistry.azurecr.io 
+            --name Contoso-Edge01-in-Wichita --repositories "contoso-app/* contoso-service/*"
+  - name: Update the next time the connected registry will sync with its parent again.
```
Please write the translation of the cron schedule in the update description so that we can provide as much info as possible to the users :)

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_params.py:
```diff
@@ -381,6 +381,20 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
+        c.argument('mode', options_list=['--mode', '-m'], type=int, help='Can be one of the two operating modes: registry or mirror.', required=False, default="registry")
+        c.argument('parent', options_list=['--parent', '-p'], type=int, help='The name of the parent ACR. If omitted, the cloud ACR will be assumed (the value from --registry).', required=False)
+        c.argument('repositories', options_list=['--repositories', '-t'], type=int, help='Specifies the repositories that need to be synched to the Connected ACR. It can include wildcards to select multiple repositories. It can be in the format [REPO01] [REPO02]...', required=True)
+        c.argument('sync_schedule', options_list=['--sync-schedule', '-s'], type=int, help='Optional parameter to define the start of the synch schedule. Uses cron expression to determine the schedule. If not specified, the instance is considered always online', required=False)
```
nit: "to define the sync schedule" instead of " to define the start of the sync schedule"

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_params.py:
```diff
@@ -381,6 +381,20 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
+        c.argument('sync_window', options_list=['--sync-window', '-w'], type=int, help='Required parameter if --sync-schedule is present. Used to determine the schedule duration. Uses ISO 8601 duration format.', required=False)
+        c.argument('disable_auto_update', options_list=['--disable-auto-update', '-a'], type=int, help='Use to disable automatically updates during synch schedule.', required=False, arg_type=get_three_state_flag())
+        c.argument('next-update', options_list=['--next-update'], type=int, help='Optional ISO 8601 timestamp parameter used to specify when the next update for the instance will happen.', required=False)
+        c.argument('log_level', options_list=['--log-level'], type=int, help='Sets the log level for logging on the instance.', required=False, default="registry")
```
Accepted log levels are Debug, Information, Warning, Error, and None. Default is Information

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_params.py:
```diff
@@ -381,6 +381,20 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
+        c.argument('next-update', options_list=['--next-update'], type=int, help='Optional ISO 8601 timestamp parameter used to specify when the next update for the instance will happen.', required=False)
+        c.argument('log_level', options_list=['--log-level'], type=int, help='Sets the log level for logging on the instance.', required=False, default="registry")
+        c.argument('sync_message_ttl', options_list=['--sync-message-ttl'], type=int, help='Determines how long the sync messages will be kept in the cloud. Uses ISO 8601 duration format.', required=False, default="P2D")
+        c.argument('disable_sync_audit_logs', options_list=['--disable-sync-audit-logs'], type=int, help='Disables synchronization of audit logs. ', required=False, arg_type=get_three_state_flag())
```
AuditLogStatus property is default to Disabled. Should this flag be `--enable-audit-logs` with a default of `false`? Our naming pattern on true/false flags uses "enabled" in the name.

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_params.py:
```diff
@@ -381,6 +381,20 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
+        c.argument('repositories', options_list=['--repositories', '-t'], type=int, help='Specifies the repositories that need to be synched to the Connected ACR. It can include wildcards to select multiple repositories. It can be in the format [REPO01] [REPO02]...', required=True)
+        c.argument('sync_schedule', options_list=['--sync-schedule', '-s'], type=int, help='Optional parameter to define the start of the synch schedule. Uses cron expression to determine the schedule. If not specified, the instance is considered always online', required=False)
+        c.argument('sync_window', options_list=['--sync-window', '-w'], type=int, help='Required parameter if --sync-schedule is present. Used to determine the schedule duration. Uses ISO 8601 duration format.', required=False)
+        c.argument('disable_auto_update', options_list=['--disable-auto-update', '-a'], type=int, help='Use to disable automatically updates during synch schedule.', required=False, arg_type=get_three_state_flag())
```
Please remove as auto update is not supported for MVP

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_params.py:
```diff
@@ -381,6 +381,20 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
+        c.argument('sync_schedule', options_list=['--sync-schedule', '-s'], type=int, help='Optional parameter to define the start of the synch schedule. Uses cron expression to determine the schedule. If not specified, the instance is considered always online', required=False)
+        c.argument('sync_window', options_list=['--sync-window', '-w'], type=int, help='Required parameter if --sync-schedule is present. Used to determine the schedule duration. Uses ISO 8601 duration format.', required=False)
+        c.argument('disable_auto_update', options_list=['--disable-auto-update', '-a'], type=int, help='Use to disable automatically updates during synch schedule.', required=False, arg_type=get_three_state_flag())
+        c.argument('next-update', options_list=['--next-update'], type=int, help='Optional ISO 8601 timestamp parameter used to specify when the next update for the instance will happen.', required=False)
```
@rosanch do you mean `--sync-schedule`?

rosanch replied:
This is one of the questions I posted in teams. I'm not sure what this flag refers to. I just added it for now to follow the CLI Experience specs https://msazure.visualstudio.com/AzureContainerRegistry/_git/functional-specs?path=%2FRegistry%2FConnected-ACR%2Fcli-experience.md&_a=preview

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_help.py:
```diff
@@ -1204,6 +1204,98 @@
+        az acr connecter-acr create --registry mycloudregistry.azurecr.io
+            --mode mirror --parent Contoso-Edge01-in-Wichita
+            --name Contoso-Mirror01-in-Wichita --repositories contoso-app/hello-world
+  - name: Create a mirror connected registry with auto update disable, that syncs every day at midninght and sync window of 4 hours.
```
For `--sync-window` and `--sync-message-ttl` the command descriptions ask for the values in ISO8601 duration format. Please update the examples to use this format. For instance, `4H` -> `PT4H`, `90D` -> `P90D`.

zhoxing-ms committed:
fix tag multiple objects (#15146)

evelyn-ys committed:
{Core} az login: Add error handling when getting msi token and parsing PEM file (#15574)

* Improvement for most frenquent errors in `az login`

fengzhou-msft committed:
refactor _arm_to_cli_mapper (#15553)

fengzhou-msft committed:
{Telemetry} Add platform information and some config values (#15608)

shurd committed:
[Cosmos DB] az cosmosdb create/update: Improve error message from incorrect --locations input (#15276)

* update error message when providing incorrect key for locations parameter

* ignore lines that are too long

* allow removing ip range filter during update

* sql analytical storage

* fixing help text typo

* fix linter

Co-authored-by: Sam Hurd <sahurd@microsoft.com>
Co-authored-by: Zunli Hu <zuh@microsoft.com>

arde0708 committed:
[RDBMS] : `az postgres|mariadb|mysql server create` : Updated create experience for Postgres, MySQL and MariaDB - new fields in the output , Introduce new values for `--public` parameter in create command (all,<IP>,<IPRange>,0.0.0.0) (#15538)

* Sterling create experience

* Updated tests for new sterling create experience

* Fix style issues

* Update Replication tests

* Updating proxyresourcetests

* Trigger checks

* Remove replication capacity validation

* Address comments

* Change positional arguments to keyword arguments

* Add sku list test results in yaml files

* Remove password as a keyword arg

* Rerun CI

* Rerun CI

* Rerun CI

* Modifying output to avoid breaking change

* Fix breaking change

* Remove preview message from sTerling commands

mjain2 committed:
[RDBMS] Add flexible-server connect command (#15419)

* Adding az <engine> flexible-server connect command

* Fixing linter and style issues

* Updating docker file for build purpose

* Addressing comments

* Updating messages for required params

* Style fix

* Adding connect tests

* Updating tests

* Fixing to generalize host name for national clouds

* Empty commit to retrigger pipeline

* Empty commit to retrigger pipeline

* Removing accidentally added env2 files

* Empty commit to trigger pipeline again

* fix CI

* Empty commit to retrigger pipeline

Co-authored-by: Zunli Hu <zuh@microsoft.com>

Juliehzl committed:
[Storage] `az storage fs access`: Support managing ACLs recursively (#15221)

* add acl recursive support

* try to ad callback

* recursive pass

* add test

* refine test but still failed

* add failed entries

* test pass for set recursive

* add update-recursive but test failed

* test pass

* refine help

* update doc

* fix style

* fix style

* test

* test pass

* test pass

* refine help

Commit:
{Release} Upgrade to Azure CLI 2.14.0 (#15637)

arrownj committed:
Revert "[RDBMS] Add flexible-server connect command (#15419)" (#15640)

This reverts commit de6522bfc46778b27a6a268903623345330aa328.

arrownj committed:
{Package} remove brew audit --strict check (#15641)

arrownj committed:
{Release} remove revert PR from HISTORY.rst (#15644)

Commit:
[Billing] Introduce new commands and upgrade to 2020-05-01 (#15568)

rosanch committed:
parameter name change for consistency

jiasli committed:
{Core} Honor scopes specified by Track 2 SDK (#15184)

rosanch committed:
pr comments

bim-msft committed:
[KeyVault] Refine error messages for HSM `list-deleted` and `purge` (#15657)

jiasli committed:
{Doc} Update doc for hotfix PR (#15552)

Co-authored-by: Feng Zhou <55177366+fengzhou-msft@users.noreply.github.com>

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_help.py:
```diff
@@ -1204,6 +1204,98 @@
+
+helps['acr connected-acr create'] = """
+type: command
+short-summary: Create a connected registry for an Azure Container Registry.
```
If the parent is an ACR, the user should not need to give us the login server. The ACR name is provided in the required `--registry` parameter as this is a proxy resource.

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_help.py:
```diff
@@ -1204,6 +1204,98 @@
+short-summary: Create a connected registry for an Azure Container Registry.
+examples:
+  - name: Create a connected registry with DNS name 'edge01.contoso.internal' that has 'mycloudregistry.azurecr.io' as a parent.
+    text: |
```
@rosanch we do not support wildcard for `--repositories`

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_params.py:
```diff
@@ -381,6 +381,20 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
+        c.argument('mode', options_list=['--mode', '-m'], type=int, help='Can be one of the two operating modes: registry or mirror.', required=False, default="registry")
+        c.argument('parent', options_list=['--parent', '-p'], type=int, help='The name of the parent ACR. If omitted, the cloud ACR will be assumed (the value from --registry).', required=False)
+        c.argument('repositories', options_list=['--repositories', '-t'], type=int, help='Specifies the repositories that need to be synched to the Connected ACR. It can include wildcards to select multiple repositories. It can be in the format [REPO01] [REPO02]...', required=True)
+        c.argument('sync_schedule', options_list=['--sync-schedule', '-s'], type=int, help='Optional parameter to define the sync schedule. Uses cron expression to determine the schedule. If not specified, the instance is considered always online', required=False)
```
You can specify that the default here is to continuously sync. The default cron is `"* * * * *"`, which is attempts the sync every minute.

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_params.py:
```diff
@@ -381,6 +381,20 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
+        c.argument('sync_window', options_list=['--sync-window', '-w'], type=int, help='Required parameter if --sync-schedule is present. Used to determine the schedule duration. Uses ISO 8601 duration format.', required=False)
+        #c.argument('auto_update_enabled', options_list=['--auto-update-enabled', '-a'], type=int, help='Use to disable automatically updates during synch schedule.', required=False, arg_type=get_three_state_flag())
+        c.argument('next_update', options_list=['--next-update'], type=int, help='Optional ISO 8601 timestamp parameter used to specify when the next update for the instance will happen.', required=False)
+        c.argument('log_level', options_list=['--log-level'], type=int, help='Sets the log level for logging on the instance. Accepted log levels are Debug, Info, Warning, Error, and None. ', required=False, default="Info")
```
`default="Information"`. This is a standard enum for log level. See https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loglevel?view=dotnet-plat-ext-3.1

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_params.py:
```diff
@@ -381,6 +381,20 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
+        c.argument('next_update', options_list=['--next-update'], type=int, help='Optional ISO 8601 timestamp parameter used to specify when the next update for the instance will happen.', required=False)
+        c.argument('log_level', options_list=['--log-level'], type=int, help='Sets the log level for logging on the instance. Accepted log levels are Debug, Info, Warning, Error, and None. ', required=False, default="Info")
+        c.argument('sync_message_ttl', options_list=['--sync-message-ttl'], type=int, help='Determines how long the sync messages will be kept in the cloud. Uses ISO 8601 duration format.', required=False, default="P2D")
+        c.argument('sync_audit_logs_enabled', options_list=['--sync-audit-logs-enabled'], type=int, help='Disables synchronization of audit logs. ', required=False, arg_type=get_three_state_flag())
```
`help= Indicates whether audit log synchronization is enabled`

jaysterp replied:
Please specify here that this value will be set to disabled if not provided.

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/connected_acr.py:
```diff
@@ -0,0 +1,192 @@
+        Schedule=sync_schedule,
+        MessageTtl=sync_message_ttl,
+        SyncWindow=sync_window,
+        LastSyncTime=None,
```
`LastSyncTime` and `GatewayEndpoint` are readonly properties, you do not need to set them to none.

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/connected_acr.py:
```diff
@@ -0,0 +1,192 @@
+    SyncProperties, ParentProperties = cmd.get_models(
+            'SyncProperties', 'ParentProperties')
+    sync_properties = SyncProperties(
+        TokenId=None, #Create Token? The resource ID of the ACR token used to authenticate the connected registry to its parent during sync. 
```
This property is required. We need to send a ACR token resource id for sync

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/connected_acr.py:
```diff
@@ -0,0 +1,192 @@
+logger = get_logger(__name__)
+
+
+def acr_connected_acr_create(cmd,
```
Please add support for a list of client token resource ids. The user will provide the list of client token resource ids, the client will not be responsible for creating the tokens (unlike sync). This corresponds to `properties.clientTokenIds`.

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/connected_acr.py:
```diff
@@ -0,0 +1,192 @@
+        Status=None,
+        Certificate=tls_certificate_properties
+    )
+    login_server_properties = LoginServerProperties(
```
`LoginServer` is a readonly property. Please remove

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/connected_acr.py:
```diff
@@ -0,0 +1,192 @@
+        ClientTokenIds=None,
+        LoginServer=login_server_properties,
+        Logging=logging_properties,
+        StatusDetails=None
```
`StatusDetails` is a readonly property. please remove

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/connected_acr.py:
```diff
@@ -0,0 +1,192 @@
+    connected_acr_create_parameters = ConnectedRegistry(
+        ProvisioningState=None,
+        Mode=mode,
+        Version=None,
```
`Version` is readonly. Please remove

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/connected_acr.py:
```diff
@@ -0,0 +1,192 @@
+        ProvisioningState=None,
+        Mode=mode,
+        Version=None,
+        LastVersionUpdateTime=None,
```
`LastVersionUpdateTime` is readonly. Please remove

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/connected_acr.py:
```diff
@@ -0,0 +1,192 @@
+        Mode=mode,
+        Version=None,
+        LastVersionUpdateTime=None,
+        ConnectionState=None,
```
`ConnectionState` is readonly. Please remove

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/connected_acr.py:
```diff
@@ -0,0 +1,192 @@
+        Version=None,
+        LastVersionUpdateTime=None,
+        ConnectionState=None,
+        LastActivityTime=None,
```
`LastActivityTime` is readonly. Please remove

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_params.py:
```diff
@@ -381,6 +381,20 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
+        c.argument('resource_group_name', options_list=['--resource-group', '-g'], help='The name of the resource group where the Connected ACR registry will be created. Must be the same as the resource group for the --registry.', required=False)
+        c.argument('registry_name', options_list=['--registry', '-r'], help='The login server of the Cloud ACR registry. Must be the FQDN to support also Azure Stack.', required=True)
+        c.argument('connected_acr', options_list=['--name', '-n'], help='Name for the Connected ACR. Name must start with a letter and contain only alphanumeric characters (including ‘_’) optionally separated by a ‘-’. 5 to 40 chars long. Name must be unique under the Cloud ACR hierarchy.', required=True)
+        c.argument('mode', options_list=['--mode', '-m'], type=int, help='Can be one of the two operating modes: registry or mirror.', required=False, default="registry")
```
Please specify that `Mirror` is a pull-only mode.

jaysterp replied:
@rosanch `type=string`

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/connected_acr.py:
```diff
@@ -0,0 +1,192 @@
+        TokenId=None, #Create Token? The resource ID of the ACR token used to authenticate the connected registry to its parent during sync. 
+        Schedule=sync_schedule,
+        MessageTtl=sync_message_ttl,
+        SyncWindow=sync_window,
```
`syncWindow` is required server side. We support a minimum sync window of `PT1H`

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_help.py:
```diff
@@ -1204,6 +1204,98 @@
+type: command
+short-summary: Delete a connected registry from Azure Container Registry.
+examples:
+  - name: Delete a mirror connected registry 'Contoso-Edge01-in-Wichita' from parent registry 'mycloudregistry.azurecr.io' and skip verification.
```
Why do we reference the cloud ACR with login server rather than resource name?

jiasli committed:
{CI} Pin to Python 3.8 for "Verify src/azure-cli/requirements.*.Windows.txt" (#15691)

rosanch committed:
pr comments

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_params.py:
```diff
@@ -381,6 +381,20 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
+        c.argument('parent', options_list=['--parent', '-p'], type=int, help='The name of the parent ACR. If omitted, the cloud ACR will be assumed (the value from --registry).', required=False)
+        c.argument('repositories', options_list=['--repositories', '-t'], type=int, help='Specifies the repositories that need to be synched to the Connected ACR. It can include wildcards to select multiple repositories. It can be in the format [REPO01] [REPO02]...', required=True)
+        c.argument('sync_schedule', options_list=['--sync-schedule', '-s'], type=int, help='Optional parameter to define the sync schedule. Uses cron expression to determine the schedule. If not specified, the instance is considered always online', required=False)
+        c.argument('sync_window', options_list=['--sync-window', '-w'], type=int, help='Required parameter if --sync-schedule is present. Used to determine the schedule duration. Uses ISO 8601 duration format.', required=False)
```
@rosanch sync_window is a required param. If you will provide a default in the CLI, please add this in the property description, similar to message ttl below

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_params.py:
```diff
@@ -381,6 +381,20 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
+        c.argument('sync_schedule', options_list=['--sync-schedule', '-s'], type=int, help='Optional parameter to define the sync schedule. Uses cron expression to determine the schedule. If not specified, the instance is considered always online', required=False)
+        c.argument('sync_window', options_list=['--sync-window', '-w'], type=int, help='Required parameter if --sync-schedule is present. Used to determine the schedule duration. Uses ISO 8601 duration format.', required=False)
+        #c.argument('auto_update_enabled', options_list=['--auto-update-enabled', '-a'], type=int, help='Use to disable automatically updates during synch schedule.', required=False, arg_type=get_three_state_flag())
+        c.argument('next_update', options_list=['--next-update'], type=int, help='Optional ISO 8601 timestamp parameter used to specify when the next update for the instance will happen.', required=False)
```
Please remove. Auto update is not supported in MVP

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_params.py:
```diff
@@ -381,6 +381,20 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
+        c.argument('registry_name', options_list=['--registry', '-r'], help='The login server of the Cloud ACR registry. Must be the FQDN to support also Azure Stack.', required=True)
+        c.argument('connected_acr', options_list=['--name', '-n'], help='Name for the Connected ACR. Name must start with a letter and contain only alphanumeric characters (including ‘_’) optionally separated by a ‘-’. 5 to 40 chars long. Name must be unique under the Cloud ACR hierarchy.', required=True)
+        c.argument('mode', options_list=['--mode', '-m'], type=int, help='Can be one of the two operating modes: registry or mirror.', required=False, default="registry")
+        c.argument('parent', options_list=['--parent', '-p'], type=int, help='The name of the parent ACR. If omitted, the cloud ACR will be assumed (the value from --registry).', required=False)
```
Why is `type=int` on these params?

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/connected_acr.py:
```diff
@@ -0,0 +1,192 @@
+                             sync_window=None,
+                             auto_update_enabled=True,
+                             next_update=None,
+                             log_level=DEFAULT_LOG_LEVEL,
```
Update should not have defaults. All params should be set to None. The user needs to specify the new values they would like to update. See task update as a sample https://github.com/Azure/azure-cli/blob/6f25ca60aa8cfd3bfea850bd802dacbd51cb214d/src/azure-cli/azure/cli/command_modules/acr/task.py#L326

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/connected_acr.py:
```diff
@@ -0,0 +1,192 @@
+                             mode=DEFAULT_MODE,
+                             sync_schedule=None,
+                             sync_window=None,
+                             auto_update_enabled=True,
```
please remove auto update params from the CLI

ThejaChoudary committed:
Commit#1346 (#15497)

jiasli committed:
{Core} Do not append/strip trailing slash in scope (#15698)

jiasli committed:
[Role] az role assignment create: Make description, condition, condition_version preview (#15690)

qwordy committed:
[Compute] Fix update permission check issue (#15606)

* [Compute] Fix update permission check issue

* Fix vmss update instance with image version

Zay2k committed:
Update check for pfx file in batch create_certificate (#15509)

Juliehzl committed:
[Storage] Fix issue #15600: az storage fs exists: in case fs does not exist ResourceNotFoundError is returned (#15643)

Juliehzl committed:
[Storage] Fix issue #15706: The examples for storage container create are incorrect (#15731)

fengzhou-msft committed:
{Test} Fix live tests for aks, advisor, appconfig, cognitiveservices, cosmosdb, dls, hdinsight, policyinsights, iot, role, security, servicefabric (#15710)

* fix aks tests

* fix advisor

* fix appconfig

* cognitiveservices

* fix cosmosdb

* fix dls tests

* fix hdinsight

* fix policyinsights tests

* fix iot tests

* fix role tests

* fix security tests

* clean up resource

* fix servicefabric tests

* revert domain change

fengzhou-msft committed:
fix cloud register example (#15655)

qwordy committed:
{Test} Pipeline (#15654)

* index.html

* history

* Commit mode condition

* fix date and time

* azdev

* azdev

* module db

* db

* db

* db

* db

* add retry

* retry

* revert

* Remove iotcentral

* Support testing Azure CLI extensions

* fix syntax error

* log

* command order

* Improve index.html. Move report link inside table.

* Display number of tests in link name.

* remove duplicate reports

* decrease maxParallel to 5

* Fix a small problem of job condition.

alexeldeib committed:
[AKS] add ephemeral os functionality (#15673)

* feat: ephemeral os functionality for aks

* fix: unused import

* fix: typo

akshayneema committed:
[BACKUP] Fix #14976: KeyError fixed and help text improved (#15718)

rosanch committed:
token logic

shenglol committed:
[ARM] Fix an issue where What-If shows two resource group scopes with different casing (#15631)

* Normalize resource scope casing

* Fix style issue

Commit:
{Misc} Update Network and Monitor codeowner (#15771)

bim-msft committed:
{KeyVault} Bump azure-mgmt-keyvault to 8.0.0 (#15714)

avanigupta committed:
{AppConfig} Stop passing resource to `get_login_credentials` method (#15736)

jiasli committed:
{Core} Allow exception_handler to mute exception (#15306)

jiasli committed:
{Core} az login: Capture managed_by_tenants error (#15550)

jiasli committed:
[Profile] az login: Add back the warning that a browser is opened (#14505)

jiasli committed:
{Core} az login: Remove the extra linebreak from warning message (#15792)

damienbod committed:
Docs functionapp create plan property requires resource id if using a plan from a different rg (#15707)

hivyas committed:
[AMS] Change "region" to "location" in output string: az ams account sp create (#15664)

houk-ms committed:
adjust the error type description (#15777)

mirdaki committed:
Update to use the appopriate User ID (#15797)

kaiqzhan committed:
Engineering improvement: replace addon strings with constants (#15786)

Co-authored-by: Kaiqi Zhang <kaiqzhan@microsoft.com>

qwordy committed:
{Compute} Update help of vm disk attach (#15768)

madsd committed:
[AppService] Fix #11698 #15198 #14862 #15409: az webapp/functionapp config access-restriction add (#15784)

* Fixing reported issues for access-restriction

* Fix help example missing parameter

karataliu committed:
[AKS] Add custom download url support for 'az aks install-cli' (#15794)

* [AKS] Add custom download url support for 'az aks install-cli'

* fixup! [AKS] Add custom download url support for 'az aks install-cli'

* fixup! fixup! [AKS] Add custom download url support for 'az aks install-cli'

qwordy committed:
{CI} Fix python version (#15804)

chshum committed:
[Security] az security pricing: Update help to reflect current API version being called (#15807)

Juliehzl committed:
{storage} Fix live test  (#15814)

rosanch committed:
Support for Gateway scope map and tokens
Logic update on connected registry update and create

detienne20 committed:
{Release} Hotfix: TS Multiline String Support for Template Inputs  (#15760)

* Multiline strings now compressed for proper JSON in template

* Updted comments

* Addressed build error

* Added test to cover hot fix

* Reverted workaround for build issue

* Nevermind

Co-authored-by: Dania Etienne <daetienn@microsoft.com>

jiasli committed:
{Core} Workaround empty and multiple scopes provided by old Track 2 SDKs (#15806)

rosanch committed:
spliting format for connected-acr list and show commands

zhoxing-ms committed:
upgrade version for hotfix (#15819)

rosanch committed:
list special format

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/connected_acr.py:
```diff
@@ -0,0 +1,298 @@
+                             registry_name,
+                             connected_acr_name,
+                             repositories,
+                             client_token_ids,
```
@rosanch `client_token_ids` is an optional property for create.

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/connected_acr.py:
```diff
@@ -0,0 +1,298 @@
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name)
+    return client.get(resource_group_name, registry_name, connected_acr_name)
+
```
@rosanch we also need a command for the deactivate API. See https://msazure.visualstudio.com/One/_git/DevServices-ContainerRegistry-Service/pullrequest/3733243

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/connected_acr.py:
```diff
@@ -0,0 +1,298 @@
+        scope_map_id = create_default_scope_map(cmd, resource_group_name, registry_name,
+                                                SYNC_SCOPE_MAP_NAME.format(connected_acr_name),
+                                                repository_actions_list, gateway_actions_list,
+                                                "Created by connected registry: {}".format(connected_acr_name))
```
Can you please update the description so customers know this token is related to syncing the connected registry?

hildurhodd committed:
[NetAppFiles] BREAKING CHANGE: Add commands for new features and updates for netappfiles 2020-06 (#15602)

qwordy committed:
[Compute] Enhancement of vm list-skus table format (#15573)

* [Compute] Refine vm list-skus table format

* update format

zhoxing-ms committed:
Merge branch 'release' of https://github.com/Azure/azure-cli into dev

qwordy committed:
[Compute] vm host group create: make --platform-fault-domain-count required and update help (#15751)

fengzhou-msft committed:
rename gpg key (#15809)

calvinsID committed:
[AppService] Fixes #15604, #15605: Add Dotnet5 support (#15721)

qwordy committed:
Release Azure CLI 2.14.2 (#15834)

shenmuxiaosen committed:
[AppConfig] Fix key vault client initialization (#15826)

fengzhou-msft committed:
reduce deb package tests (#15833)

jim-minter committed:
[ACS] add v3 deprecation warnings (#15802)

houk-ms committed:
add new error types (#15838)

houk-ms committed:
{Core} Remove useless codes in core for ai-did-you-mean-this (#15793)

* remove useless codes for ai-did-you-mean-this

* remove recommendation_provider

* suppress error message caused by extension not installed

* no longer need failure recommendation test in parser

Commit:
[Billing] Add new command groups to manage Agreement, RoleAssignment, RoleDefinition and Instructions (#15697)

houk-ms committed:
{Error Handling} Developer guidelines to onboad the new error output (#15774)

* add developer doc to onboard the new error output

* fix typos

* fix typos

* refine the doc format

* fix typos

* Fix typos

Co-authored-by: Feiyue Yu <iamyfy@163.com>
Co-authored-by: Bin Ma <bim@microsoft.com>

* Apply suggestions from code review

Co-authored-by: Jiashuo Li <jiasli@microsoft.com>

* Apply suggestions from code review

Co-authored-by: Jiashuo Li <jiasli@microsoft.com>

* update new error types

* add contact for adding new error type

* fix typos

Co-authored-by: Feiyue Yu <iamyfy@163.com>
Co-authored-by: Bin Ma <bim@microsoft.com>
Co-authored-by: Jiashuo Li <jiasli@microsoft.com>

Commit:
[Network] Fix bug that can't create a Standard_v2 application-gateway without a private static IP address (#15757)

Zhaph committed:
{Storage} Fix #15842: az storage blob delete-batch typo in docs (#15843)

rosanch committed:
List client tokens command
List utility upgrades
Pr comments addressed

evelyn-ys committed:
ignore `--auth-mode` argument for `az queue policy` command groups (#15815)

gzuber committed:
Added Node 14. Removed preview flag from custom handlers. (#15840)

jiasli committed:
{Core} Refactor client factory for Track 2 SDK (#15791)

jiasli committed:
{Misc.} Trim trailing whitespace in .github/pull_request_template.md (#15850)

Commit:
[Network] az network dns zone import: Raise FileOperationError instead of FileNotFoundError if zone file doesn't exist (#15870)

jiasli committed:
{Core} Add comments to prevent calling AdalAuthentication._token_retriever (#15854)

evelyn-ys committed:
{Core} az login: Add error handling when finding subscriptions (#15776)

* Add error handling when finding subscriptions

bim-msft committed:
[KeyVault] Support selective key restore for managed HSMs (#15867)

arde0708 committed:
{RDBMS} Introduce tests for Vnet/Subnet and public-access in Flexible-server commands. (#15830)

* Adding more tests for vnet and public access scenarios

* Use VirtualNetworkPreparer instead of manually creating virtual networks

jsntcy committed:
[Core] Upgrade azure-mgmt-core to 1.2.1 (#15780)

kaiqzhan committed:
[AKS] az aks browse: point to Azure Portal Kubernetes resources view if k8s >=1.19 or kube-dashboard not enabled (#15803)

gzuber committed:
[AppService] functionapp: Added warning if customer tries to switch functionapp from premium to consumption. (#15695)

* Added warning if customer tries to switch functionapp from premium to consumption.

* Added --force flag. Added error if not a windows to windows migration. Changed exception handling.

* Fixed style errors.

* Changed Error types to be more descriptive.

Commit:
[Privatedns] az network private-dns zone import: Raise FileOperationError instead of FileNotFoundError if zone file doesn't exist  (#15874)

anusapan committed:
Add missing tags option in dps create command (#15648)

aim-for-better committed:
[HDInsight] Add two parameters `--resource-provider-connection` and `--enable-private-link` to support relay outbound and private link (#15837)

bim-msft committed:
{Core} Fix the incorrect token expiration on ADAL tokens (#15875)

houk-ms committed:
{Error Improvement} Adjust error types (#15865)

* adjust error types

* adjust error handling guideline

* refine error handling guidelines

* replace UnknownError

* Update .github/pull_request_template.md

Co-authored-by: Yishi Wang <yishiwang@microsoft.com>

Co-authored-by: Yishi Wang <yishiwang@microsoft.com>

jiasli committed:
{ARM} az account management-group show: Refine help for --name (#15871)

aim-for-better committed:
{Core} Add US Gov cloud endpoint for synapse (#15881)

* {Docs} Remove stale reference in README to closed issue about extensions (#12771)

* Add endpoint and resource id for UsGov cloud, and change the resource id for mooncake

Co-authored-by: Daniel Miller <daniel.keegan.miller@gmail.com>
Co-authored-by: Azure CLI Team <AzPyCLI@microsoft.com>
Co-authored-by: Zhenyu Zhou <zhezhou@microsoft.com>

bim-msft committed:
{Core} Fix the incorrect usage `full_token['expires_on']` (#15880)

evelyn-ys committed:
{Storage} Add error handling for file related exceptions (#15775)

* fix '_NoneType_ object is not iterable'

* add file related exception handler

* add recommendations

fengzhou-msft committed:
{Doc} Update instructions on using airgapped clouds (#15754)

fengzhou-msft committed:
use the more general JMESPathError (#15873)

qwordy committed:
{Test} Live test 0.8.0 (#15789)

* Optimize index.html CSS style

* Support a permanent link of latest testing results.

* fix a html label error

arrownj committed:
[Compute] Support update vm/image version when they use cross tenant images (#15883)

* [Compute] Support update vm/image version when they use cross tenant images

jsntcy committed:
[Storage] Migrate from track1 to track2 (#15845)

* [Storage] Migrate from track1 to track2

* fix test failure

* Revert method name in storage module due to extension depending on it

* fix test error

* fix test error

* fix tests

* fix test

* fix test failure

* fix test failure

* Fix storage track2 package for CI

* add import statement in patch script

rosanch committed:
PR comments
Update and Create corner cases
minor fixes

Juliehzl committed:
{Storage} Fix live test (#15885)

norshtein committed:
[AKS] Support BYO control plane identity (#15862)

calvinsID committed:
[AppService] Fixes #15604, #15605: Add Dotnet5 support (#15721) (#15898)

amsoedal committed:
[AKS] az aks use-dev-spaces: Indicate that dev-spaces commands are deprecated (#15785)

* indicate that dev-spaces commands are deprecated

* make text uppercase

* follow documentation

* fix lines that are too long

* remove typo

* remove line endings

* pr feedback

Commit:
[Network] Fix NoneType error crash while deleting nonexisting resources of ApplicationGateway, LoadBalancer, Nic (#15886)

zhoxing-ms committed:
{ARM} Fix live test (#15782)

* Fix live test

* Fix CI

* Fix the check for system separator

* Fix CI style

zhoxing-ms committed:
Print out error details for deployment (#15658)

jiasli committed:
Merge pull request #15903 from Azure/release

{Branch Management} Merge release 2.14.2 back to dev

Commit:
{Release} Upgrade to Azure CLI 2.15.0 (#15902)

jsntcy committed:
Update HISTORY.rst (#15904)

fengzhou-msft committed:
{Test} Fix live tests for appconfig, batchai, advisor, make some tests in container, security record_only (#15887)

Commit:
{Network} Fix live test that caused by there is no self.skip() method (#15905)

jiasli committed:
{Role} Make test_role_assignment_audits live run more reliable (#15197)

HappyTobi committed:
[Network] az network network watcher configure: Fix NetworkWatcherCountLimitReached error (#15801)

Fix NetworkWatcherCountLimitReached error caused by case sensitivity of location value

Commit:
[Network] az network application-gateway http-listener: Fix bug that cannot create and update with WAF policy name (#15929)

Juliehzl committed:
{Storage} `az storage account file-service-properties update`: fix the scenario when no property is specified (#15928)

* fix issue

* test pass

* add test

* generoic update

* test pass

dkorzuno committed:
[ARO] CLI enhancements, including extension suppression (#15897)

akshayneema committed:
[BACKUP] Fix #14976: CLI error improvements for ValueError and AttributeError cases (#15861)

Commit:
[aks] az aks nodepool add/update/upgrade: takes max surge parameter (#15740)

* mirror max surge/commit 32ee9feaa up from az extenstions

* indent wrong

* fix up style in max surge additions

* help in wrong place

houk-ms committed:
failure recommendation enhancement (#15841)

bim-msft committed:
[KeyVault] Add a parameter `--include-managed` for `az keyvault key/secret list` to list managed resources (#15926)

rosanch committed:
Show/list table format update

jiasli committed:
{Core} Vendor track 2 subscription SDK in azure-cli-core (#15711)

Commit:
[NetAppFiles] Update azure-mgmt-netapp version to 0.14.0 (#15934)

timothyta12 committed:
[Search] Add SKU option (#15895)

rosanch committed:
connected-acr refactor to connected-registry

Commit:
{Network} Update certificate files to fix live test test_vpn_client_package (#15982)

calvinsID committed:
[AppService] Bugfix #14857: Let users update webapp config even with access restriction (#15945)

* Let users update webapp config even with webapp access restriction

* Added tests

evelyn-ys committed:
[Profile] Hotfix: Fix #15961: az login: UnboundLocalError: local variable 'token_entry' referenced before assignment (#15962)

jiasli committed:
{Release} HotFix: Release 2.15.1 (#15992)

arde0708 committed:
{RDBMS} Update test servers to be public access by default (#15955)

akshayneema committed:
[BACKUP] az backup protection undelete: Added support for AzureWorkload protection undelete using CLI (#15979)

* added undelete functionality for azure workload type

* Change ralted to error handling guide

Co-authored-by: Akshay Neema <akneema@microsoft.com>

handonMS committed:
[MySQL] Add MySQL version upgrade cli  (#15977)

akshayneema committed:
Fix for Bad Request Error in item list and policy list case (#15999)

Co-authored-by: Akshay Neema <akneema@microsoft.com>

con-f-use committed:
{Packaging} Add required python version in various setup.py (#15719)

jiasli committed:
Merge branch 'release' into dev

fengzhou-msft committed:
{Deb Install} Add support for more systems and user provided dist name (#15998)

Juliehzl committed:
{Storage} Make sure default kind is Storage in Azure Stack (#16035)

Commit:
Update description for kek arg (#15866)

a-santamaria committed:
[ServiceFabric] Update SF app docs. only support for arm deployed resources (#16003)

* update app docs. only support for arm deployed resources

* also for versions

* fix typo

* Align verb form with manage

Co-authored-by: Alfredo Santamaria Gomez <alsantam@microsoft.com>

rosanch committed:
new command connected-registry install
sync-token-id option for create and update commands
cli experience design revisions

akshaysngupta committed:
[AKS] Add support for AGIC addon (#15993)

yonzhan committed:
[Network] Deprecate route table V1 (#16039)

Juliehzl committed:
{Storage} Remove max api for azure stack (#16054)

Hazhzeng committed:
[AppService] Allow Azure Functions Create to accept Python 3.9 parameters (#16020)

* Add Azure Functions Python 3.9 support

* Change Python 3.9 to Preview and Hidden

* Python 3.9 shouldn't be the default

qwordy committed:
{Test} Live test pipeline update (#16042)

* Update index.html and update email content

* Clean policy automatically

* Fix no log output captured problem

jagilber committed:
fix broken link. commit https://github.com/MicrosoftDocs/azure-docs/commit/67dcbf8818752f0013019605945bdef5ea668136 (#16062)

fengzhou-msft committed:
update homebrew template and build with update_existing (#16068)

nexxai committed:
Typo fixes (#15916)

jiasli committed:
{CI} Remove usage of six (#16082)

yungezz committed:
mark az upgrade preview (#16036)

Jeej committed:
[APIM] az apim restore: New command to restore a backup of an API Management service (#15933)

jiasli committed:
{Role} az ad app permission add: Add instruction on how to get available permission IDs (#15849)

Commit:
{CI} Pin pip==20.2.4 and fix CredScan upgrade (#16097)

arde0708 committed:
{RDBMS} Update error message if user passes server name in upper case (#16108)

houk-ms committed:
{Error Improvement} Error category and error message refining (#15963)

* suppress UnknownError in error message

* categorize HttpResponse 4xx errors

* refine error message for HttpOperationError

* code style refining

dbradish-microsoft committed:
[Role] Add long-summary/warning for commands generating credentials (#15825)

jeffwmartinez committed:
Update _help.py with typo and example (#16103)

removing an extra word "with". Adding an example for creating a Windows container app service plan with the new PV3 SKU.

jiasli committed:
{Core} Show custom headers in --debug log (#15635)

fore5fire committed:
[CDN] Add multi-origin preview and BYOC autorotation. (#15348)

qwordy committed:
{Test} Fix cognitiveservices and batch test (#16058)

* {Test} Fix cognitiveservices

* Fix batch

* Fix cognitiveservices

* Fix batch

Juliehzl committed:
update codeowner (#16114)

DaeunYim committed:
[RDBMS] Add single server show-connection-string and tests for local-context commands, server creation  (#15844)

* test recordings and style linter

* Trigger CI

* Trigger CI

* adding missing command

* fix mysql connect and host names

* style issue fixed

* test fixed

* test fix

* test fix

* Trigger CI

rosanch committed:
scope map action refactor: messages -> message

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/_utils.py:
```diff
@@ -469,3 +453,120 @@ def prepare_source_location(cmd, source_location, client_registries, registry_na
     """
+
+
+# Scope & Tolens help functions
```
nit: spelling

rosanch committed:
deactivate command

jdmartinez36 committed:
{Misc} Remove '?view=' syntax in links where not necessary (#16107)

rosanch committed:
install command upgrade

houk-ms committed:
{Error Improvement} Error handling guidelines for extensions (#16116)

* extension guideline for error handling

* Update doc/error_handling_guidelines.md

Co-authored-by: Feiyue Yu <iamyfy@163.com>

Co-authored-by: Feiyue Yu <iamyfy@163.com>

fengzhou-msft committed:
add ubuntu groovy package (#16132)

aim-for-better committed:
[Synapse] Support synapse sql dw cmdlets and update az synapse workspace create cmdlet (#16095)

jsntcy committed:
[Network] `az network cross-region-lb`: Support cross-region load balancer (#16131)

* [Network] Support cross-region load balancer

* Fix linter error

* Re-recording to fix test failure

* Fix linter errors

* Fix linter error

* Fix test

norshtein committed:
[AKS] Change MSI cluster to default (#16057)

* Change MSI cluster to default

* Fix lint

* Fix lint 2

* Fix indent

* Manually edit recording file to pass integration test

* Fix ingress appgw error

* Revert "Manually edit recording file to pass integration test"

This reverts commit 39b5b76e86726231ed6b599cdf15ac10f36075fd.

* Not prompt

* Fix typo

* Add failed test case back

* Make ingress appgw test live only

* Fix virtual node role assignment error

* Revert "Fix virtual node role assignment error"

This reverts commit 0066427f47d2fe27c00398e2495b7ee1c7c3c7f3.

Commit:
[Network] az network express-route port generate-loa: New commands to generate and download the PDF letter of authorization for a ExpressRoutePort (#16135)

kairu-ms committed:
{monitor} Fix Non actionable issues Phase 1 (#15882)

* [monitor] use InvalidArgumentValueError instead of CliError and ValueError

* throw valueError in get_period_type

* remove 'usage error:' in error messages

* fix bug in monitor_exception_handler

* remove monitor customed exception_handlers use default handle_exception instead.

* simplify monitor exception handler

jiasli committed:
{Misc.} Print "Error detail" for SSLError  (#16024)

fengzhou-msft committed:
{Upgrade} az upgrade: Handle NoTTYException and fix extension upgrade for homebrew (#16083)

abbhar committed:
[Core] Add 2020-09-01-hybrid API profile for AzureStack (#15096)

* Adding 2020-09-01-hybrid API profile for AzureStack

* fixing Data_private_Keyvault & MGMT_resource_template loading issue

* Fixing the CIfailure issue

* adding disk access version

* added gallary & resource feature versions

* adding storage,IOT & KV tests suppresion

* fixing code integrity failure

* reverting KV to older version

* vm kv, sa, testcase fixes

jorgecotillo committed:
removed preview flag and bump api version (#16019)

kairu-ms committed:
[monitor] az monitor metrics alert support dynamic thresholds (#15820)

* az monitor metrics alert support dynamic thresholds

* Add license headers.

* Fix antlr parse errors

* fix clint errors

* add command 'monitor metrics alert dimension create' and command 'monitor metrics alert condition create'

* Add validators and examples for `metrics alert condition create` and `metrics alert dimension create`

* add help for condition

* add test for `monitor metrics alert condition create`

* fix style issues

* fix lint issues

* fix test_metric_alert_condition_create

* remove unused argument 'client'

* rename parameter --window to --evaluation.

* use 'Include' as default value for parameter '--operator' for command 'monitor metrics alert dimension create'

* Rename --violation and --evaluation by--num-violations and --num-periods

* rollback command monitor log-analytics workspace support no wait

* remove type=str in parameters

* rename _metric_alert_dimension_prefix and add preview tag for dimension create and condition create commands

calvinsID committed:
[AppService] Fixes #16041: az webapp config ssl create results in unknown error (#16124)

* Fix bug:az webapp config ssl create results in unknown error

* Use 2 minute timeout, add comment to use LongRunningOperation once it is supported

* Use URL returned in header instead of certificate GET call for polling

* added webapp config ssl show command, and updated error msg to use show instead of list

* add functionapp config ssl show as well

jsntcy committed:
{Misc} Add missing recording file (#16147)

olayemio committed:
{Compute} Update shared image gallery scenarios (#15958)

* updated 'az sig image-version create/update'

* included custom image scenario for az vmss update

* correct custom image scenario for az vmss update

* remove trailing whitespace

* Delete launch.json

* restore launch.json

* fixed changes to launch.json

* copied original launch.json

* checked out from main remote

Commit:
{Release} Upgrade to Azure CLI 2.16.0 (#16152)

* update azure-cli version to 2.16.0

* Apply suggestions from code review

* Apply suggestions from code review

* Apply suggestions from code review

Co-authored-by: Xing Zhou <Zhou.Xing@microsoft.com>

Co-authored-by: Yu Chen <ychenu@microsoft.com>
Co-authored-by: Xing Zhou <Zhou.Xing@microsoft.com>

c-ryan-k committed:
Updated key-generation (#16129)

Co-authored-by: Ryan Kelly <rykelly@microsoft.com>

gingi committed:
[Batch] azure-batch 10.0.0 / API v2020-09-01.12.0 (#16156)

* Batch data plane (azure-batch) upgraded to 10.0.0
* `az batch job task-counts` now returns a complex object including both task counts as well as task slot counts
* Updated Batch testing README

gingi committed:
Update CODEOWNERS (#16161)

Updates Batch code owners

evelyn-ys committed:
{Storage} err improvement (#16034)

* error handling supplement

* add unit test

akshaysngupta committed:
[AKS] fix: add arguments removed by a previous PR (#16080)

JohnRusk committed:
[AKS ] az aks get-credentials: Clarify documentation for get-credentials (#16011)

Juliehzl committed:
[Storage] Fix issue #15965: Clarify how to remove multiple legal hold tags with `az storage container legal-hold [clear|set]`  (#16167)

kairu-ms committed:
{monitor} Fix monitor test 5/7 (#15876)

* add spring cloud build for test_monitor_autoscale_rule_for_spring_cloud

* update ext_sub for test_monitor_clone_storage_metric_alerts_across_subs_scenario

* clean log profiles before create in test_monitor_create_log_profile

* clean diagnostic-settings before create in test_monitor_diagnostic_settings_subscription

* fix test_monitor_log_analytics_workspace_linked_storage

* record test_monitor_autoscale_rule_for_spring_cloud

jiasli committed:
{Misc.} Update CODEOWNERS (#16171)

jiasli committed:
{Core} Honor expiresOn, instead of expiresIn (#15901)

Juliehzl committed:
{azure-cli-testsdk} add location parameter for VirtualNetworkPreparer (#16173)

houk-ms committed:
get status code for HttpOperationError (#16110)

qwordy committed:
[Compute] New license type RHEL_ELS_6 (#16012)

* [Compute] New license type RHEL_BYOS_ELS_6

* update

yungezz committed:
update codeowners (#16176)

Commit:
{CI} Enable --no-exitfirst for automation test in CI (#16190)

qwordy committed:
[Compute] Adopt track2 SDK, azure-mgmt-compute==18.0.0 (#15750)

* [Compute] azure-mgmt-compute==17.0.0

* Fix test_vm_disk_upload, test_vm_managed_disk

* fix 13 tests

* Fix test

* fix more test

* Fix more test

* Fix more test

* Fix more test

* Fix more test

* Fix style

* Fix delete

* Compute 18.0.0

* Fix more test

* image begin_delete

* image begin_create_or_update

* image begin_create_or_update

* snapshot begin_remoke_access

* Add begin_

* vm host begin_create_or_update

* gallery, image-definition, begin_create_or_update

* Fix test in hybrid-2019-03-01

* Fix test of hybrid-2018-03-01

* Fix network test

* Fix backup

* Fix backup

* Fix backup test

* Fix backup test

* Fix backup test

* Fix network test

* patch_models.py

* remove debug code

* Fix test_disable_encryption_error_cases_handling

* Fix test_disable_encryption_error_cases_handling

* Fix style

* auxiliary token

* Fix test_disable_encryption_error_cases_handling in hybrid_2020_09_01

* Fix test_disable_encryption_error_cases_handling in hybrid_2020_09_01

* fix test in vm in hybrid_2020_09_01

rosanch committed:
Bug Fixes and streamlining the udpate command

arde0708 committed:
{RDBMS} Update location against which tests are executed (#16158)

* Update error message if user passes server name in upper case

* Update error message returned from API

* Update target location for tests

* Update Failing tests

* Address PR comments

msyyc committed:
bugfix for help of network (#16174)

ehashman committed:
[ARO] az aro delete: Add RBAC validation on cluster deletion (#16101)

rosanch committed:
upgrading acr sdk version to rc16

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/connected_registry.py:
```diff
@@ -0,0 +1,480 @@
+            'Update ambiguity. Duplicate client token ids were provided with ' +
+            '--add-client-token-ids and --remove-client-token-ids arguments.\n{}'.format(errors))
+
+    if sync_message_ttl is None:
```
@rosanch why are we resetting this? If the user does not wish to set during update this it can be null

jaysterp commented on src/azure-cli/azure/cli/command_modules/acr/connected_registry.py:
```diff
@@ -0,0 +1,480 @@
+
+    if sync_message_ttl is None:
+        sync_message_ttl = current_connected_registry.parent.sync_properties.message_ttl
+    ConnectedRegistryUpdateParameters, SyncProperties, LoggingProperties = cmd.get_models(
```
@rosanch please import `SyncUpdateProperties` model for PATCH instead of `SyncProperties`. `SyncProperties` is used for PUT.

fengzhou-msft committed:
record test_aro_delete (#16226)

fengzhou-msft committed:
{Test} Fix or skip live tests for cdn, privatedns, reservations (#16221)

rosanch committed:
cleanup and minor fixes

rosanch committed:
use token names instead of ids for connected registry

rosanch committed:
Merge branch 'dev' into roy/connected-acr

rosanch commented:
ready for upstream fork
